Library/Homebrew/cask/artifact/moved.rb

```diff
@@ -34,39 +35,53 @@ def summarize_installed
 
       private
 
-      def move(adopt: false, force: false, verbose: false, command: nil, **options)
+      def move(adopt: false, force: false, verbose: false, predecessor: nil, reinstall: false,
+               command: nil, **options)
         unless source.exist?
           raise CaskError, "It seems the #{self.class.english_name} source '#{source}' is not there."
         end
 
         if Utils.path_occupied?(target)
-          if adopt
-            ohai "Adopting existing #{self.class.english_name} at '#{target}'"
-            same = command.run(
-              "/usr/bin/diff",
-              args:         ["--recursive", "--brief", source, target],
-              verbose:      verbose,
-              print_stdout: verbose,
-            ).success?
-
-            unless same
-              raise CaskError,
-                    "It seems the existing #{self.class.english_name} is different from " \
-                    "the one being installed."
+          if !predecessor.nil? && target.directory? && target.children.empty? && predecessor.artifacts.any? do |a|
+               a.class.dsl_key == self.class.dsl_key && a.target == target
```

Comment:
```suggestion
               a.class == self.class && a.target == self.target
```

Reply:
> doesn't like the redundant `self` 

Weird, because above it's using that redundant `self` as well.

Reply:
In that case it allows `self.target` because there's also a local variable named `target`. There isn't a local `target` variable here.